roms_tools/setup/cdr_release.py

```diff
@@ -0,0 +1,481 @@
+from dataclasses import dataclass
+from abc import abstractmethod, ABC
+from pydantic import BaseModel, model_validator, Field, ConfigDict
+from typing import List, Dict, Union, Literal, Optional
+from typing_extensions import Annotated
+from annotated_types import Ge, Le
+from datetime import datetime
+import warnings
+from roms_tools.setup.utils import get_tracer_defaults
+
+NonNegativeFloat = Annotated[float, Ge(0)]
+
+
+@dataclass
+class ValueArray(ABC):
+    """Abstract base class representing a time series of values, either scalar or list.
+
+    Attributes
+    ----------
+    name : str
+        Name of the variable (e.g., flux or concentration).
+    values : Union[float, List[float]]
+        A constant value or a time-varying series of values.
+    """
+
+    name: str
+    values: Union[float, List[float]]
+
+    def check_length(self, num_times: int):
```

Comment:
-> None

Reply:
done in [849833e](https://github.com/CWorthy-ocean/roms-tools/pull/301/commits/849833e8291e7c1a17c3246129fbaf71e1f34fcd)